code/datums/components/material_container.dm

```diff
@@ -129,8 +129,21 @@
 		to_chat(user, span_warning("[I] does not contain sufficient materials to be accepted by [parent]."))
 		return
 	if(!has_space(material_amount))
-		to_chat(user, span_warning("[parent] is full. Please remove materials from [parent] in order to insert more."))
-		return
+		if(istype(I, /obj/item/stack))
+			//figure out how much space is left
+			var/space_left = max_amount - total_amount
+			//figure out the amount of sheets that can fit that space
+			var/obj/item/stack/stack_to_split = I
+			var/material_per_sheet = material_amount / stack_to_split.amount
+			var/sheets_to_insert = round(space_left / material_per_sheet)
+			if(!sheets_to_insert)
+				to_chat(user, span_warning("[parent] can't hold any more of [I] sheets."))
+				return
+			//split the amount we don't need off
+			stack_to_split.split_stack(user, stack_to_split.amount - sheets_to_insert)
```

Comment:
Your CI is failing because of this proc-call here.
`split_stack` calls `put_in_hands` which sleeps, and the proc itself should not sleep.
You may need to invoke `split_stack` asynchronously. (`INVOKE_ASYNC`).